### PR TITLE
Add e2e tests

### PR DESCRIPTION
**Pull request by wama-tw**
- Add e2e tests:
  - home page
    - click NCU+
    - click 登入
    - click Features 大量選課心得
    - cover background image change
    - hover 全校課程
    - click 全校課程 in dropdown
    - click 心得總覽 in dropdown
    - click 模擬排課 in dropdown
  - courses page
    - type 課名搜尋
    - type 課號搜尋
    - type 老師搜尋

### CODE REVIEW

**Comment by wama-tw** [comment]
TODO: #133

File: backend/src/seeder/course-seeder/course-seeder.service.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    const course = new Course();
+    course.year = 111;
+    course.semester = 0;
+    course.serialNo = 1001;
+    course.classNo = 'SE6030*';
+    course.title = '軟體工程實務';
+    course.credit = 1;
+    course.passwordCard = 'passwordCard';
+    course.teachers = ['["梁德容", "王尉任", "鄭永斌"]'];
+    course.classTimes = ['["1-2", "1-3", "1-4"]'];
```
Comment (ppodds) [inline backend/src/seeder/course-seeder/course-seeder.service.ts:24]: from
```
course.teachers = ['["梁德容", "王尉任", "鄭永斌"]'];
course.classTimes = ['["1-2", "1-3", "1-4"]'];
```
to
```
course.teachers = '["梁德容", "王尉任", "鄭永斌"]';
course.classTimes = '["1-2", "1-3", "1-4"]';
```

File: cypress.config.ts [hunk 2]
```diff
@@ -6,5 +6,6 @@ export default defineConfig({
     setupNodeEvents(on, config) {
       // implement node event listeners here
     },
+    baseUrl: "http://localhost:3001",
```
Comment (ppodds) [inline cypress.config.ts]: Why not use 4000?

**Review by ppodds — CHANGES_REQUESTED** [review]

File: backend/src/seeder/course-seeder/course-seeder.service.ts (hunk 1 above)
Comment (wama-tw) [reply backend/src/seeder/course-seeder/course-seeder.service.ts:24]: But the data type of course.teachers and course.classTimes are string[]

File: cypress.config.ts (hunk 2 above)
Comment (wama-tw) [reply cypress.config.ts]: It had been changed in this commit: https://github.com/NCU-Plus/ncuplus/pull/132/commits/08c883a7a9b45a262cc075ecf9f10731397c6069

File: backend/src/app.module.ts [hunk 3]
```diff
@@ -31,7 +31,7 @@ import { ReportModule } from './report/report.module';
     CollegeModule,
     DepartmentModule,
     CourseModule,
-    AuthModule,
```
Comment (ppodds) [inline backend/src/app.module.ts:34]: You can't just disable `AuthModule`. It would cause serious problems.

**Review by ppodds — CHANGES_REQUESTED** [review]

File: backend/src/seeder/course-seeder/course-seeder.service.ts (hunk 1 above)
Comment (ppodds) [reply backend/src/seeder/course-seeder/course-seeder.service.ts:24]: The export type of `Course` is wrong. Just correct it to `string`. Thanks!

**Pull request closed without merging**